faster_sam/middleware.py

```diff
@@ -5,10 +5,49 @@
 
 
 class RemovePathMiddleware(BaseHTTPMiddleware):
+    """
+    Middleware for removing a specified part of the request path.
+
+    e.g
+
+    >>> app = FastAPI()
+    >>> app.add_middleware(RemovePathMiddleware, path="/foo")  # Removes /foo from requests path
+    >>> # /foo/bar ---> midleware ----> /bar
+    >>> @app.get("/bar")
+    ... def bar():
+    ...     return {"message": "Responding to GET /foo/bar}
+    """
+
     def __init__(self, app: ASGIApp, path: str) -> None:
+        """
+        Initializes the RemovePathMiddleware.
+
+        Parameters
+        ----------
+        app : ASGIApp
+            The application that the middleware is applied.
```

Comment:
O que você acha de trocar a expressão "aplicar" por "registrar"?

Me parece mais apropriado dizer que registramos um conjunto de funções intermediárias.

_Application instance the middleware is being registered to._